firebase-common/src/main/java/com/google/firebase/heartbeatinfo/HeartBeatInfoStorage.java

```diff
@@ -50,15 +75,66 @@ static synchronized HeartBeatInfoStorage getInstance(Context applicationContext)
     return instance;
   }
 
+  synchronized void storeHeartBeatInformation(String heartBeatTag, long millis) {
+    this.heartBeatSharedPreferences.edit().putString(String.valueOf(millis), heartBeatTag).apply();
+    this.heartBeatCount += 1;
+    if (this.heartBeatCount > HEART_BEAT_COUNT_LIMIT) {
+      this.cleanUpStoredHeartBeats();
+    }
+  }
+
+  private synchronized void cleanUpStoredHeartBeats() {
+    ArrayList<Long> timestampList = new ArrayList<>();
+    for (Map.Entry<String, ?> entry : heartBeatSharedPreferences.getAll().entrySet()) {
+      timestampList.add(Long.parseLong(entry.getKey()));
+    }
+    Collections.sort(timestampList);
+    for (Long millis : timestampList) {
+      this.heartBeatSharedPreferences.edit().remove(String.valueOf(millis)).apply();
+      this.heartBeatCount -= 1;
+      if (this.heartBeatCount <= (HEART_BEAT_COUNT_LIMIT / 2)) return;
+    }
+  }
+
+  synchronized long getLastGlobalHeartBeat() {
+    return sharedPreferences.getLong(GLOBAL, -1);
+  }
+
+  synchronized void updateGlobalHeartBeat(long millis) {
+    sharedPreferences.edit().putLong(GLOBAL, millis).apply();
+  }
+
+  synchronized List<SdkHeartBeatResult> getStoredHeartBeats(boolean shouldClear) {
+    ArrayList<SdkHeartBeatResult> sdkHeartBeatResults = new ArrayList<>();
+    for (Map.Entry<String, ?> entry : heartBeatSharedPreferences.getAll().entrySet()) {
+      long millis = Long.parseLong(entry.getKey());
+      String sdkName = ((String) entry.getValue());
+      sdkHeartBeatResults.add(SdkHeartBeatResult.create(sdkName, millis));
+    }
+    Collections.sort(sdkHeartBeatResults);
+    if (shouldClear) clearStoredHeartBeats();
+    return sdkHeartBeatResults;
+  }
+
+  void clearStoredHeartBeats() {
+    heartBeatSharedPreferences.edit().clear().apply();
+  }
+
+  boolean isValidHeartBeat(long base, long target) {
```

Comment:
looking at the signature, iiuc this method does not have anything to do with heartbeat, but rather checks if 2 timestamps belong to the same day in utc, consider renaming the method to something like

```suggestion
  boolean static isSameDateUtc(long oneMillis, long otherMillis) {
```

not `static`, since you don't use any member state of this class.